config/config.go

```diff
@@ -1066,6 +1066,8 @@ type Config struct {
 
 	// ResourceSync configures mitigation strategy in case sync fails.
 	ResourceSync ResourceSyncConfig `json:"resource_sync"`
+
+	Private Private `json:"-"`
```

Comment:
If you use embedding as suggested, you will not have to refactor the references across /gateway, making the change smaller

Reply:
since this is completely internal/private, it's okay to do that refactoring to make it explicit that it's private. right?